pkg/storage/utils/decomposedfs/node/locks.go

```diff
@@ -307,3 +312,24 @@ func (n *Node) hasLocks(ctx context.Context) bool {
 	_, err := os.Stat(n.LockFilePath()) // FIXME better error checking
 	return err == nil
 }
+
+func isLockModificationAllowed(oldLock *provider.Lock, newLock *provider.Lock) (bool, error) {
+	if oldLock.Type == provider.LockType_LOCK_TYPE_SHARED {
+		return true, nil
+	}
+
+	var allowed bool = true
```

Comment:
Can we reverse the logic and default this to `false`? 
It's good practice to have permissions default to `false` because in case of a logic error in the function (maybe through later refactoring) the possibility of unauthenticated access is minimized.

Reply:
done :+1: